### PR TITLE
pubsub: new publish snippet

### PR DESCRIPTION
**Pull request by pongad**
the subscriber side soon to follow

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
LGTM, but wait for @lesv to review and don't merge before Tue PST.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  /**
+   * Example of publishing messages.
+   */
+  // [TARGET publish(PubsubMessage)]
+  // [VARIABLE "my_project_name"]
+  // [VARIABLE "my_topic_name"]
+  public void publish(String projectName, String topicName) throws Exception {
+    // [START publish]
+    Publisher publisher = Publisher.newBuilder(TopicName.create(projectName, topicName)).build();
+    List<String> messages = Arrays.asList("message1", "message2");
```
Comment (lesv) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java]: You might wish to mention that Publisher is a very long lived thing.  I'd imaging that Publisher and message list would be what's passed to the function (perhaps from some other function), and the activity just does it's thing.  You could even call it from the loop as I don't see how the loop adds anything useful to the example.

**Comment by pongad** [comment]
@lesv PTAL. I'm not 100% sure I understand you correctly, but please let me know if this looks better. I also wanted to make sure the sample for creating the publisher also shuts it down. There're lots of threads to leak if you forget.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  public PublisherSnippets(Publisher publisher) {
+    this.publisher = publisher;
+  }
+
+  /**
+   * Example of publishing a message.
+   */
+  // [TARGET publish(PubsubMessage)]
+  // [VARIABLE "my_message"]
+  public void publish(String message) {
```
Comment (lesv) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java]: Looks a lot better.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java [hunk 3]
```diff
@@ -635,6 +622,30 @@ private Builder(TopicName topic) {
      * <p>For performance, this client benefits from having multiple channels open at once. Users
      * are encouraged to provide instances of {@code ChannelProvider} that creates new channels
      * instead of returning pre-initialized ones.
+     *
+     * <p>Example of creating a {@code Publisher}.
```
Comment (lesv) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java]: This looks good, though it might be in the wrong place ??

**Comment by pongad** [comment]
@lesv PTAL. Looks like the snippet-insertion script doesn't work well with one-line comments or something.

**Comment by garrettjonesgoogle** [comment]
LGTM with the understanding that tests will come next

**Pull request merged**